pkg/reconciler/events/auditlogs/resources/filterbuilder.go

```diff
@@ -0,0 +1,64 @@
+package resources
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	keyPrefix   = "protoPayload"
+	methodKey   = keyPrefix + ".methodName"
+	serviceKey  = keyPrefix + ".serviceName"
+	resourceKey = keyPrefix + ".resourceName"
+	typeKey     = keyPrefix + ".\x22@type\x22"
+	typeValue   = "type.googleapis.com/google.cloud.audit.AuditLog"
+)
+
+// Stackdriver query builder for querying audit logs. Currently
+// supports querying by the AuditLog serviceName, methodName, and
+// resourceName.
+type FilterBuilder struct {
+	serviceName  string
+	methodName   string
+	resourceName string
+}
+
+func (fb *FilterBuilder) SetServiceName(serviceName string) {
+	fb.serviceName = serviceName
+}
+
+func (fb *FilterBuilder) SetMethodName(methodName string) {
+	fb.methodName = methodName
+}
+
+func (fb *FilterBuilder) SetResourceName(resourceName string) {
+	fb.resourceName = resourceName
+}
```

Comment:
nit: "With" functions might be easier to use.

Reply:
Done.